core/iwasm/aot/aot_loader.c

```diff
@@ -4035,6 +4035,17 @@ create_sections(AOTModule *module, const uint8 *buf, uint32 size,
                     }
                 }
             }
+            else {
+                if (alloc_mem_for_section_body) {
+                    section->section_body = loader_malloc(
+                        section->section_body_size, error_buf, error_buf_size);
```

Comment:
Should also modify `destroy_sections`, which don't free section_body now (only munmap section_body when it is aot text and destroy_aot_text is true). Had better add a new argument `bool free_section_body`, and pass `alloc_mem_for_section_body` to it in this function:
```C
static void
destroy_sections(AOTSection *section_list, bool free_section_body,
                 bool destroy_aot_text)
```